R/geom-rug.R

```diff
@@ -157,5 +157,44 @@ GeomRug <- ggproto("GeomRug", Geom,
 
   draw_key = draw_key_path,
 
-  rename_size = TRUE
+  rename_size = TRUE,
+
+  handle_na = function(self, data, params) {
+    sides_aes <- character()
+
+    if (grepl("b|t", params$sides)) {
+      sides_aes <- c(sides_aes, "x")
+    }
+
+    if (grepl("l|r", params$sides)) {
+      sides_aes <- c(sides_aes, "y")
+    }
+
+    if (length(sides_aes) > 0) {
+      df_list <- lapply(
+        sides_aes,
+        function(axis) {
+          remove_missing(
+            data, params$na.rm,
+            c(axis, self$required_aes, self$non_missing_aes),
+            snake_class(self)
+          )
+        }
+      )
+      data <- switch(
+        paste0(sides_aes, collapse = ""),
+        "x" = ,
+        "y" = df_list[[1]],
+        "xy" = dplyr::union(df_list[[1]], df_list[[2]])
```

Comment:
ggplot2 doesn't use dplyr (it is in suggests because for some cross-functionality).
I think a similar thing could be achieved by using `vec_set_union()` from {vctrs}.

Reply:
Thanks - I have updated this as suggested